server/src/api/v1.py

```diff
@@ -755,17 +784,18 @@ def get_agents_on_queue(queue_name):
 
 
 def generate_token(allowed_resources, secret_key):
-    """Generates JWT token with queue permission given a secret key"""
+    """
+    Generates JWT token with queue permission given a secret key
+    See retrieve_token for more information on the contents of
+    the token payload
+    """
     expiration_time = datetime.utcnow() + timedelta(seconds=2)
     token_payload = {
         "exp": expiration_time,
         "iat": datetime.now(timezone.utc),  # Issued at time
         "sub": "access_token",
     }
-    if "max_priority" in allowed_resources:
-        token_payload["max_priority"] = allowed_resources["max_priority"]
-    if "allowed_queues" in allowed_resources:
-        token_payload["allowed_queues"] = allowed_resources["allowed_queues"]
+    token_payload.update(allowed_resources)
```

Comment:
Do you think it might be a good idea to add a `permissions` field to the payload, next to `exp`, `iat` and `sub`, and then have the "allowed resources" live under that? I feel it might add a little more structure to the payload and also allow for the future addition of things like scopes or roles.